### DISCUSSION TITLE
Optimizing Data Extraction from RFEM Models

### DISCUSSION THREAD

**Discussion by Jeroen124** [Dlubal-Software/RFEM_Python_Client#339] [Q&A]
Hello RFEM Developers/Community,

I'm currently working on a project using RFEM, and I've been extracting data from my model by looping over all results, following the approach demonstrated in the "timber_hall.py" example. I'm wondering if there's a more efficient method to perform this task.

### Specifically, I'm interested in knowing:
- Is there a way to extract data from multiple membersets or load combinations at once, rather than individually?
- Are there any recommended practices or methods for bulk data extraction to optimize performance?

- Currently I am running two for loops over this piece of code. Which is not ideal and takes a lot of time. I must be missing something?
```                   
ResultsTable = ResultTables.MembersInternalForcesByMemberSet(
                        loading_type = CaseObjectType.E_OBJECT_TYPE_LOAD_COMBINATION, 
                        loading_no = loadcombination.pk,
                        object_no = MemberSet.pk)
```

Any guidance or suggestions you could provide would be greatly appreciated.

Thank you for your time and assistance.

Best regards,
Jeroen

**Comment by heetrojivadiya** [comment]
Hello @Jeroen124, 

unfortunately, we do not have filtering options for class ResultTables and their static methods but alternatively you can extract multiple data at once by exporting results. We have already developed [export result tables methods](https://github.com/Dlubal-Software/RFEM_Python_Client/blob/main/RFEM/ImportExport/exports.py). You may use `ExportResultTablesToCSV()` or `ExportResultTablesWithDetailedMembersResultsToCSV()` methods and can extract all the result tables data at once which speed up your data extraction process. Later on you can easily manupulate with csv files and read your desired results. Also, you may use `ExportToTables()` method to get all the data at once in excel spreadsheets.

<img width="434" alt="image" src="https://github.com/Dlubal-Software/RFEM_Python_Client/assets/108512789/58498ab8-d4e5-4336-a85d-d0ec04a09b71">

**Comment by Jeroen124** [comment]
Hi @heetrojivadiya, as discussed in https://github.com/Dlubal-Software/RFEM_Python_Client/issues/343 we are trying to speed up the process. As models grow larger, exporting to CSV gives us 580 unique CSV files, it can not be the case that it will be faster to first export everything to CSV and later go through all these files with Pandas right? 

Some improved static methods would help, for a certain Loadcombination, i would like to extract a memberset, and get all its maximal values (Maybe seperate this for Strenght and Stiffness checks. e.g. a function to export all extreme Local Deformations and a function to export all extreme Internal forces).
